### PR TITLE
Fix #6723 for postgres and add migration model

### PR DESCRIPTION
**Pull request by SuperTux88**

### CODE REVIEW

File: lib/diaspora/federated/shareable.rb [hunk 1]
```diff
@@ -101,18 +101,18 @@ def update_existing_sharable(shareable)
...
+        def receive_shareable_visibility(user, shareable)
+          user.receive_shareable(shareable)
           user.notify_if_mentioned(shareable)
           logger.info "event=receive payload_type=#{self.class} status=complete " \
-                      "sender=#{diaspora_handle} receiver=#{person.diaspora_handle} guid=#{shareable.guid}"
+                      "sender=#{diaspora_handle} receiver=#{user.diaspora_handle} guid=#{shareable.guid}"
         end
 
-        def receive_non_persisted(user, person)
+        def receive_non_persisted(user)
```
Comment (diaspora-code-review) [inline lib/diaspora/federated/shareable.rb]: Assignment Branch Condition size for receive_non_persisted is too high. [23.43/20]

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -148,6 +148,22 @@
           Post.for_visible_shareable_sql(Time.now + 1, "created_at")
         end
 
+        context "with two posts with the same timestamp" do
+          before do
+            aspect_id = alice.aspects.where(name: "generic").first.id
+            Timecop.freeze Time.now do
```
Comment (diaspora-code-review) [inline spec/models/post_spec.rb]: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -148,6 +148,22 @@
...
+              alice.post(:status_message, text: "first", to: aspect_id)
+              alice.post(:status_message, text: "second", to: aspect_id)
+            end
+          end
+
+          it "returns them in reverse creation order" do
+            posts = Post.for_visible_shareable_sql(Time.now + 1, "created_at")
```
Comment (diaspora-code-review) [inline spec/models/post_spec.rb]: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

File: db/migrate/20160225232049_link_share_visibilities_with_user.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
+class LinkShareVisibilitiesWithUser < ActiveRecord::Migration
+  class ShareVisibility < ActiveRecord::Base
+  end
+
+  def up
```
Comment (diaspora-code-review) [inline db/migrate/20160225232049_link_share_visibilities_with_user.rb]: Method has too many lines. [23/20]

File: db/migrate/20160225232049_link_share_visibilities_with_user.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    add_index :share_visibilities, :user_id
+    add_index :share_visibilities, %i(shareable_id shareable_type user_id), name: :shareable_and_user_id
+    add_index :share_visibilities, %i(shareable_id shareable_type hidden user_id),
+              name: :shareable_and_hidden_and_user_id
+
+    add_foreign_key :share_visibilities, :users, name: :share_visibilities_user_id_fk, on_delete: :cascade
+  end
+
+  def down
```
Comment (diaspora-code-review) [inline db/migrate/20160225232049_link_share_visibilities_with_user.rb]: Method has too many lines. [21/20]

**Comment by denschub** [comment]
Thanks.

**Pull request merged**